### PR TITLE
Connect with deal cards

### PR DESCRIPTION
**Pull request by xtrembaker**

### CODE REVIEW

File: client/src/bootstrap/selectors.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-export function getAppLoadedStatus(state) {
-   return state.bootstrap;
+export default function getAppLoadedStatus(state) {
```
Comment (brissaur) [inline client/src/bootstrap/selectors.js]: nope

File: client/src/card/card.view.js [hunk 2]
```diff
@@ -1,30 +1,15 @@
...
-            src={getCardSrc(suit, cardValue)}
-            alt={getCardName(suit, cardValue)}
+            src={getCardSrc(card.id)}
+            alt={card.id}
             style={styles.card}
         />
     );
 }
-
-Card.propTypes = {
```
Comment (brissaur) [inline client/src/card/card.view.js]: !!!!!!

File: client/src/card/sagas.js [hunk 3]
```diff
@@ -11,11 +11,11 @@ function* cardPlayed({ card }) {
...
     const snap = yield document.get();
     yield document.update({
-        trick: [...snap.data().trick, card.id],
+        trick: [...snap.data().trick, card],
         // TODO: add an object with card.id && player.id
     });
 }
 
-export function* watchers() {
+export default function* watchers() {
```
Comment (brissaur) [inline client/src/card/sagas.js]: idem

File: client/src/hand/hand.view.js [hunk 4]
```diff
@@ -18,9 +18,5 @@ export default function Hand({ cards }) {
 }
 
 Hand.propTypes = {
-    cards: PropTypes.arrayOf(PropTypes.shape({
-        suit: PropTypes.string,
-        cardValue: PropTypes.string,
-        id: PropTypes.number,
-    })),
+    cards: PropTypes.arrayOf(PropTypes.string),
```
Comment (brissaur) [inline client/src/hand/hand.view.js]: not coherent with props in component Card
If it is a string like "AH", then it should not be named cards

File: client/src/table/helpers.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line import/prefer-default-export
+export const filterPlayer = (playerId, payload) => {
+    const playerFound = payload.players.filter(player => (playerId === player.id));
```
Comment (brissaur) [inline client/src/table/helpers.js]: why not use [].find

File: client/src/table/sagas.js [hunk 6]
```diff
@@ -22,10 +27,57 @@ export function* createTableAndAddPlayerToTable() {
...
+/**
+ * @doc https://github.com/redux-saga/redux-saga/blob/master/docs/advanced/Channels.md#using-the-eventchannel-factory-to-connect-to-external-events
+ * @param document
+ * @return DocumentSnapshot
+ */
+function createSnapshotChannel(document) {
+    // `eventChannel` takes a subscriber function
+    // the subscriber function takes an `emit` argument to put messages onto the channel
+    return eventChannel((emit) => {
+        const snapshotHandler = (snap) => {
```
Comment (brissaur) [inline client/src/table/sagas.js]: this is equivalent to `const snapshotHandler = emit`

File: client/src/table/sagas.js [hunk 7]
```diff
@@ -22,10 +27,57 @@ export function* createTableAndAddPlayerToTable() {
...
 
     yield put(setTableId(document.id));
 }
+
+/**
+ * Hold action on snapshot event trigger
+ * @param payload
+ */
+function* incomingSnapshotData(payload) {
+    const player = filterPlayer(yield select(getPlayerId), payload);
```
Comment (brissaur) [inline client/src/table/sagas.js]: !

File: client/src/table/sagas.js [hunk 8]
```diff
@@ -22,10 +27,57 @@ export function* createTableAndAddPlayerToTable() {
...
+    if (player.cards) {
+        yield put(dealCardsToPlayer(player.cards));
```
Comment (brissaur) [inline client/src/table/sagas.js]: updateHand ?

File: client/src/table/sagas.js [hunk 9]
```diff
@@ -22,10 +27,57 @@ export function* createTableAndAddPlayerToTable() {
...
     });
 
     yield put(setTableId(document.id));
 }
+
+/**
+ * Hold action on snapshot event trigger
+ * @param payload
+ */
+function* incomingSnapshotData(payload) {
```
Comment (brissaur) [inline client/src/table/sagas.js]: name does not mean mluch :'-(

File: client/src/bootstrap/sagas.js [hunk 10]
```diff
@@ -1,12 +1,13 @@
...
 import { registerPlayer } from '../player/sagas';
-import { createTableAndAddPlayerToTable } from '../table/sagas';
+import { createTableAndAddPlayerToTable, watchOnSnapshot } from '../table/sagas';
 import { setAppLoaded } from '../bootstrap/duck';
 
 export default function* initAppSagas() {
     try {
         yield call(registerPlayer);
         yield call(createTableAndAddPlayerToTable);
+        yield call(watchOnSnapshot);
```
Comment (brissaur) [inline client/src/bootstrap/sagas.js]: I don't think this should be called here at the bootstrap level. IMO, it should be ran each time we "join" a table: we want to be notified of what happpens (and idem, if we were to leave it, we should unsubscribe)
WDYT?

File: client/src/bootstrap/selectors.js (hunk 1 above)
Comment (xtrembaker) [reply client/src/bootstrap/selectors.js]: OK

File: client/src/card/card.view.js (hunk 2 above)
Comment (xtrembaker) [reply client/src/card/card.view.js]: OK

File: client/src/bootstrap/sagas.js (hunk 10 above)
Comment (xtrembaker) [reply client/src/bootstrap/sagas.js]: Oui, je ne savais vraiment pas où le mettre....

**Pull request merged**